ui/src/variables/utils/hydrateVars.test.ts

```diff
@@ -0,0 +1,295 @@
+// Utils
+import {ValueFetcher} from 'src/variables/utils/ValueFetcher'
+import {hydrateVars} from 'src/variables/utils/hydrateVars'
+
+// Types
+import {Variable} from '@influxdata/influx'
+import {CancellationError} from 'src/types/promises'
+
+const createVariable = (
+  name: string,
+  query: string,
+  selected?: string
+): Variable => ({
+  name,
+  id: name,
+  orgID: 'howdy',
+  selected: selected ? [selected] : [],
+  arguments: {
+    type: 'query',
+    values: {
+      query,
+      language: 'flux',
+    },
+  },
+})
+
+class FakeFetcher implements ValueFetcher {
+  responses = {}
+
+  fetch(_, __, query, ___, ____, _____) {
+    return {
+      cancel: () => {},
+      promise: this.responses[query],
+    }
+  }
+
+  setResponse(variable, response) {
+    this.responses[variable.arguments.values.query] = response
+  }
+}
+
+describe('hydrate vars', () => {
+  test('should invalidate cyclic subgraphs', async () => {
+    // Construct the following graph:
+    //
+    //     digraph {
+    //       a -> b
+    //       b -> c
+    //       c -> d
+    //       d -> e
+    //       d -> b
+    //       f -> g
+    //     }
```

Comment:
This is Graphviz syntax. There's a variety of different [editors](https://dreampuf.github.io/GraphvizOnline) you can plug this into to get a pretty graph:

![graph](https://user-images.githubusercontent.com/638955/53999233-8c82e880-40f7-11e9-91bd-df7925001a7a.png)